civiform/civiform
# Adding accessiblity to accordion per accessibility audit

### Description

Following WCAG [2.1.1. Keyboard](https://www.w3.org/TR/WCAG21/#keyboard), adding the ability to operate the accordion widget through the keyboard only.  Also, made the accordion focusable.  The accordion now conveys role (e.g. button) and state (expanded/collapsed).  The accordion is used in three places:  static text questions (in the application flow), static text question preview (in admin question creation) and in the API docs help section.

## Release notes

Following WCAG [2.1.1. Keyboard](https://www.w3.org/TR/WCAG21/#keyboard), adding the ability to operate the accordion widget through the keyboard only.  Also, made the accordion focusable.  The accordion now conveys role (e.g. button) and state (expanded/collapsed).  The accordion is used in three places:  static text questions (in the application flow), static text question preview (in admin question creation) and in the API docs help section.

### Checklist

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)


#### User visible changes



### Instructions for manual testing

1. From the homepage, click "Apply" on the "Comprehensive Sample Program".
2. On the summary page, click "Continue"
3. Turn a screen reader on.
4. Tab through the first questions page till you get to the "What are the Eligibility Requirements?" accordion.  Once there, note the focus box around the accordion.
5. Also, notice that the screen reader announces that you are on a collapsed button and reads the button text.
6. Click "Enter" or the space-bar to expand or collapse the accordion.


![image](https://github.com/civiform/civiform/assets/134556723/6533062d-0835-42ae-abf4-49032c6335f3)



### Issue(s) this completes

Fixes #3560
